### ISSUE TITLE
Add debug logging

### ISSUE THREAD

**Issue by CosminPerRam** [gamedig/rust-gamedig#188]
**What is this feature about?**
Debug logs are very useful in examining behavior in all kinds of possible use cases.
This helps in eventual user-encountered bugs.

**Additional context/references**
Add debug logging like in the Node version:
[
![image](https://github.com/gamedig/rust-gamedig/assets/7972857/6f1edb61-d415-4f93-8f96-f59385f3d58d)
](https://github.com/gamedig/node-gamedig/blob/cf0144683dbcb0a62b271ffbeffbee0ba4fbb604/lib/GlobalUdpSocket.js#L26)
(just a simple example of logging something)

The [log](https://crates.io/crates/log) crate would be perfect for this, I don't know any better alternatives.

**Comment by cainthebest** [comment]
Is this for just debug logging as we could just add a cfg assertion and print it.

**Comment by cainthebest** [comment]
eg

```rust
fn debug_print<T: std::fmt::Debug>(msg: T) {
    if cfg!(debug_assertions) {
        println!("{:?}", msg);
    }
}

fn main() {
    debug_print("This will only be printed in debug mode");
}
```

**Comment by Douile** [comment]
> the log crate would be perfect for this

There's also tracing but that is a lot more complex.

**Comment by CosminPerRam** [comment]
> eg

I wouldn't use something like this as the user couldn't control this with their log facade.
Many libraries already use the log create for debug and stuff.

**Comment by cainthebest** [comment]
Debug logging has been implemented on #217 with log, it will be further refined as more is implemented. Will close this for now as completed
